fsmonitor.c

```diff
@@ -162,26 +189,56 @@ void refresh_fsmonitor(struct index_state *istate)
 	 * should be inclusive to ensure we don't miss potential changes.
 	 */
 	last_update = getnanotime();
```

Comment:
Just a thought:  `getnanotime()` has a bunch of slop in it.  For example, the first call to it within the process always gets a value ending with `000` on Windows (observed when I started trace2).  And there's some other mysterious magic in there that I don't completely understand.

We might want to consider just getting the whole seconds since epoch, subtract 1 or 2, and then convert that to nanoseconds for storage in the header.  And err on the side of caution.